### DISCUSSION TITLE
Enquiries as regard to the storage factory

### DISCUSSION THREAD

**Discussion by Jizzie10** [Cyfrin/foundry-full-course-cu#1354] [General]
Im a newbie in the space in lesson 3. while using version 0.8.23 why does it give an error when i try to use the evm(merge) and not the evm version: shanghai

**Comment by ghost** [comment]
Each EVM version represents a hard fork (really cool topic).

A hard fork is a radical breaking change to the protocol;  features can be added or removed and that affect how smart contracts run on the blockchain. When a hard fork occurs backward compatibility may not be guaranteed, therefore, it's important to verify that the compiler and EVM versions are compatible.

If you want to compile on the Merge EVM version then you need an older solc version.

**Reply by Jizzie10** [reply]
Alright, i appreciate the insight to this.
